AsyncDisplayKit/ASViewController.m

```diff
@@ -0,0 +1,47 @@
+//
+//  ASViewController.m
+//  AsyncDisplayKit
+//
+//  Created by Huy Nguyen on 16/09/15.
+//  Copyright (c) 2015 Facebook. All rights reserved.
+//
+
+#import "ASViewController.h"
+#import "ASAssert.h"
+#import "ASDimension.h"
+
+@implementation ASViewController
+
+- (instancetype)initWithNode:(ASDisplayNode *)node
+{
+  if (!(self = [super init])) {
+    return nil;
+  }
+  
+  ASDisplayNodeAssertNotNil(node, @"Node must not be nil");
```

Comment:
You can now modify your method signature + parameters to indicate that `node` must not be nil by doing the following:
`- (instancetype)initWithNode:(ASDisplayNode *_Nonnull)node` ( XCode 7+)
`- (instancetype)initWithNode:(ASDisplayNode * nonnull)node` (Xcode 6.4)

https://developer.apple.com/swift/blog/?id=25